org.eclipse.jdt.core.tests.model/src/org/eclipse/jdt/core/tests/model/ResolveTests18.java

```diff
@@ -3126,4 +3131,47 @@ public void testBug546563() throws Exception {
 		elements
 	);
 }
+
+// Inspired from ReslveTests18.test0027
+public void test0027_BindingForLambdaMethod() throws JavaModelException {
+	this.wc = getWorkingCopy(
+			"/Resolve/src/X.java",
+			"""
+			interface I {
+			    I doit(I xyz);
+			}
+			public class X {
+				public static void main(String[] args) {
+					I i = (pqr) -> {
+						return (xyz) -> {
+							return (abc) -> abc;
+						};
+					};
+				}
+			}
+			""");
+
+	String str = this.wc.getSource();
+	String selection = "abc";
+	int start = str.lastIndexOf(selection);
+	int length = selection.length();
+
+	ASTParser parser = ASTParser.newParser(AST.getJLSLatest());
+	parser.setSource(this.wc);
+	parser.setProject(this.wc.getJavaProject());
+	parser.setWorkingCopyOwner(this.wc.getOwner());
+	parser.setCompilerOptions(this.wc.getOptions(true));
+	parser.setResolveBindings(true);
+	parser.setBindingsRecovery(true);
+	parser.setStatementsRecovery(true);
+	CompilationUnit dom = (CompilationUnit)parser.createAST(null);
+	Name variable = (Name)new NodeFinder(dom, start, length).getCoveredNode();
+	IJavaElement javaElement = variable.resolveBinding().getJavaElement();	
+
+	assertElementsEqual(
+		"Unexpected elements",
+		"abc [in doit(I) [in <lambda #1> [in doit(I) [in <lambda #1> [in doit(I) [in <lambda #1> [in main(String[]) [in X [in [Working copy] X.java [in <default> [in src [in Resolve]]]]]]]]]]]]",
+		new IJavaElement[] { javaElement }
+	);
```

Comment:
At every level lambda ordinal is \#1 - is that right ??

Reply:
It's a similar assertion that's implemented in test0027, with `1` being the ordinal on every level (unlike anonymous which have an ordinal based on the including type because it's used to generate the .class, lambdas do not generate a .class and thus ordinal seems only relative to the direct parent)